File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if (nums.length == 0) return 0;
       HashSet<Integer> set = new HashSet<>();
       for(int num:nums) set.add(num);
       int longest =1;
       for(int num: nums ){

           if(!set.contains(num-1)){ 
                int currLongest =1;
                while(set.contains(num + 1)){
                    num++;
                    currLongest++;
                }
                longest = Math.max(longest, currLongest);
                
           }

       }
       return longest;
    }
}

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.go
func longestConsecutive(nums []int) int {
	set := make(map[int]struct{}, len(nums))
	for _, n := range nums {
		set[n] = struct{}{}
	}

	longest := 0

	for n := range set {
		if _, ok := set[n-1]; ok {
			continue
		}

		length := 1
		for m := n + 1; ; m++ {
			if _, ok := set[m]; !ok {
				if length > longest {
					longest = length
				}
				break
			}
			length++
		}
	}

	return longest
}
